File: Aigilas/Aigilas/States/MainMenuState.cs
using SPX.States;
using SPX.Entities;
using Aigilas.Creatures;
using Aigilas.Dungeons;
using SPX.Text;
using SPX.Util;
using SPX.Core;
using Aigilas.Management;
using System;
using Aigilas.IO;
using SPX.IO;

namespace Aigilas.States
{
    public class MainMenuState : State
    {
        private ActionTextHandler _text = new ActionTextHandler();

        private const string PlayText = "Play Game";
        private const string OptionsText = "Options";
        private const string QuitText = "Quit Game";
        private const string SelectionText = "--<";

        private int _selection;

        public MainMenuState()
        {
            Input.SetContext(Contexts.Nonfree, Client.Get().GetFirstPlayerIndex());
        }
        public void Update()
        {
            _text.WriteAction(PlayText, 1, 300, 100);
            _text.WriteAction(OptionsText, 1, 300, 200);
            _text.WriteAction(QuitText, 1, 300, 300);

            _selection += (Input.IsActive(Commands.MoveDown, Client.Get().GetFirstPlayerIndex()) ? 1 : 0)
                + (Input.IsActive(Commands.MoveUp, Client.Get().GetFirstPlayerIndex()) ? -1 : 0);
            _selection %= 3;
            if (_selection < 0)
            {
                _selection = 0;
            }

            if (Client.Get().IsGameStarting())
            {
                for (int ii = 0; ii < Client.Get().GetPlayerCount(); ii++)
                {
                    Input.SetContext(Contexts.Free, ii);
                }            
                StateManager.LoadState(new GameplayState());
            }
            else
            {
                if (Input.IsActive(Commands.Confirm, Client.Get().GetFirstPlayerIndex()))
                {
                    switch (_selection)
                    {
                        case 0:
                            Client.Get().StartGame();
                            return;
                        case 1:
                            StateManager.LoadState(new OptionsState());
                            return;
                        case 2:
                            Environment.Exit(0);
                            return;
                        default: break;
                    }
                }
            }

            _text.WriteAction(SelectionText,1,225,100*(_selection+1));
        }
        public void LoadContent()
        {

        }
        public void Draw()
        {
            _text.Draw();
            _text.Clear();
        }
    }
}

File: aigilas/src/aigilas/states/MainMenuState.java
package aigilas.states;

import aigilas.net.Client;
import aigilas.ui.SelectableButton;
import aigilas.ui.UiAssets;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
import org.apache.commons.io.FileUtils;
import sps.bridge.Contexts;
import sps.bridge.Sps;
import sps.core.Loader;
import sps.core.Logger;
import sps.graphics.Assets;
import sps.graphics.Renderer;
import sps.io.Input;
import sps.states.StateManager;

import java.io.File;

public class MainMenuState extends MenuState {
    public MainMenuState() {
        super();
        Label.LabelStyle lblStyle = new Label.LabelStyle(Assets.get().font(), Color.WHITE);
        Label title = new Label("Aigilas", lblStyle);

        final SelectableButton startGameBtn = new SelectableButton("Start Local", UiAssets.getButtonStyle());
        startGameBtn.addListener(new ChangeListener() {
            public void changed(ChangeEvent event, Actor actor) {
                Logger.info("Starting the game");
                Client.get().startGame();
            }
        });

        SelectableButton startServerBtn = new SelectableButton("Host LAN", UiAssets.getButtonStyle());
        startServerBtn.addListener(new ChangeListener() {
            public void changed(ChangeEvent event, Actor actor) {
                StateManager.loadState(new StartHostServerState());
            }
        });

        SelectableButton connectToServerBtn = new SelectableButton("Join LAN", UiAssets.getButtonStyle());
        connectToServerBtn.addListener(new ChangeListener() {
            public void changed(ChangeEvent event, Actor actor) {
                StateManager.loadState(new JoinServerAsGuestState());
            }
        });

        SelectableButton helpBtn = new SelectableButton("Help", UiAssets.getButtonStyle());
        helpBtn.addListener(new ChangeListener() {
            public void changed(ChangeEvent event, Actor actor) {
                StateManager.loadState(new HelpMenuState());
            }
        });

        SelectableButton optionsBtn = new SelectableButton("Options", UiAssets.getButtonStyle());
        optionsBtn.addListener(new ChangeListener() {
            public void changed(ChangeEvent event, Actor actor) {
                StateManager.loadState(new OptionsState());
            }
        });

        SelectableButton exitBtn = new SelectableButton("Exit", UiAssets.getButtonStyle());
        exitBtn.addListener(new ChangeListener() {
            public void changed(ChangeEvent event, Actor actor) {
                Gdx.app.exit();
            }
        });

        add(startGameBtn);
        add(startServerBtn);
        add(connectToServerBtn);
        table.row();
        add(helpBtn);
        add(optionsBtn);
        add(exitBtn);

        title.setY((int) (Renderer.get().VirtualHeight * .8));
        title.setX((int) (Renderer.get().VirtualWidth * .4));
        stage.addActor(title);

        File versionDat = Loader.get().data("version.dat");
        if (versionDat.exists()) {
            try {
                String versionText = FileUtils.readFileToString(versionDat);
                Label version = new Label("Version: " + versionText, lblStyle);
                stage.addActor(version);
            }
            catch (Exception e) {
                Logger.info("WARNING: No version information found at " + versionDat.getAbsolutePath());
            }
        }
        startGameBtn.setSelected(true);
    }

    @Override
    public void update() {
        super.update();
        if (Client.get().isGameStarting()) {
            for (int ii = 0; ii < Client.get().getPlayerCount(); ii++) {
                Input.get().setContext(Contexts.get(Sps.Contexts.Free), ii);
            }
            StateManager.loadState(new LoadingState());
        }
    }

    @Override
    public String getName() {
        return "MainMenuState";
    }
}
